designdocs/provider_contexts.md

```diff
@@ -0,0 +1,231 @@
+This may be desirable if there is a subsidiary company or other complex structures.
+
+#### User Trust in Application Presentation
+
+Users need to know what application they are logging into and trust that the displayed information is not a phishing attack.
+Users have a reasonable expectation that the chain and wallet provide a level of protection against phishing attacks from Providers and Applications.
+
+#### Easy Wallet Integration of Data
+
+Wallets displaying login request information need to be able to easily obtain and verify the information they want to show the user.
+
+## Proposal
+
+Frequency should provide a safeguarded system of setting and updating Provider and Provider Application Contexts for wallets to display to users.
+
+- [Storage and Structure of the Data](#data)
+- [Mainnet Approval Flow](#governance)
+- [Example of Wallet Usage via SIWF](#siwf)
+- [Provider Dashboard Steps](#dashboard)
+
+### **Storage and Structure of the Data** <a id='data'></a>
+
+Currently, the `ProviderRegistryEntry` has only a `name` property, and no way to update said name.
+
+Applications have two different pieces of default and (potentially) internationalized data:
+- name
+- logo/image
+
+Limits
+1. Provide a space to internationalize up to 150 localizations (Windows and macOS each have < 150).
+2. Image MUST BE encoded (`png`)[https://www.w3.org/TR/png-3/] that is `250x100` (support for future sizes and light/dark should be considered).
+
+The data structure must support both of these and internationalization.
+
+Proposed are the following changes:
+
+1. `ProviderRegistryEntry` struct be renamed `ApplicationRegistryEntry`
+2. `ApplicationRegistryEntry` have the following properties:
+    ```rust
+    pub struct ApplicationRegistryEntry<T, U>
+    where
+        T: Get<u32>,
+        U: Get<u32>,
+    {
+        pub default_name: BoundedVec<u8, T>,
+        pub localized_names: BTreeMap<Vec<u8>, BoundedVec<u8, T>>,
+        pub default_logo_250_100_png_bytes: BoundedVec<u8, U>,
+        pub localized_logo_250_100_png_bytes: BTreeMap<Vec<u8>, BoundedVec<u8, U>>,
+    }
+    ```
+3. The `ProviderToRegistryEntry` be updated to use `ProviderToApplicationRegistryEntry`
+    ```rust
+    #[pallet::storage]
+	pub type ProviderToRegistryEntry<T: Config> = StorageMap<
+		_,
+		Twox64Concat,
+		ProviderId,
+		ProviderToApplicationRegistryEntry<T::MaxProviderNameSize, T::MaxProviderLogo250X100Size>,
+		OptionQuery,
+	>;
+    ```
+4. New `ProviderToApplicationRegistryEntry` storage be initialized:
+    ```rust
+    // Alias for clarity
+    type ApplicationIdentifier<T: Config> = BoundedVec<u8, T::MaxProviderNameSize>
+
+    #[pallet::storage]
+	pub type ProviderToApplicationRegistryEntry<T: Config> = StorageDoubleMap<
+		_,
+		Twox64Concat,
+		ProviderId,
+		Twox64Concat,
+		ApplicationIdentifier<T>,
+		ApplicationRegistryEntry<T::MaxProviderNameSize, T::MaxProviderLogo250X100Size>,
+		OptionQuery,
+	>;
+	```
```

Comment:
This looks confusing to me. Can a `StorageMap` reference another StorageMap? Seems like there's some naming mixups.

Also, I realize that `ApplicationIdentifier` is not in the critical path anywhere, but should we keep identifiers normalized (ie, numeric/integer)?